### ISSUE TITLE
Error while setting up battery_notes platform for binary_sensor|sensor|button

### ISSUE THREAD

**Issue by karco** [andrew-codechimp/HA-Battery-Notes#934]
## System Information

version | core-2024.2.1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.1
os_name | Linux
os_version | 6.1.74-haos
arch | x86_64
timezone | America/Toronto
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4376
Installed Version | 1.34.0
Stage | running
Available Repositories | 1386
Downloaded Repositories | 36

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | December 2, 2024 at 19:00
relayer_connected | true
relayer_region | us-east-1
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | true
remote_server | us-east-1-0.ui.nabu.casa
certificate_status | ready
instance_id | 4b99e2919fbe46b6be53bc9b383af956
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.5
-- | --
update_channel | stable
supervisor_version | supervisor-2024.01.1
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 30.8 GB
disk_used | 13.6 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Samba share (12.2.0), Terminal & SSH (9.8.1), MariaDB (2.6.1), InfluxDB (5.0.0), Z-Wave JS UI (3.3.2), Grafana (9.1.3), phpMyAdmin (0.9.1), Network UPS Tools (0.13.0), Studio Code Server (5.15.0), Zigbee2MQTT (1.35.3-1), Mosquitto broker (6.4.0), Let's Encrypt (5.0.15), ESPHome (2023.12.9), Hydroqc Add-on (v1.2.0-1), Advanced SSH & Web Terminal (17.1.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 3
-- | --
resources | 18
views | 36
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | January 30, 2024 at 20:46
-- | --
current_recorder_run | February 9, 2024 at 16:02
estimated_db_size | 583.11 MiB
database_engine | mysql
database_version | 10.6.12

</details>

### Checklist

- [X] I have added `battery_notes:` to my configuration.yaml and restarted.
- [X] I have read the [FAQ's](https://andrew-codechimp.github.io/HA-Battery-Notes/faq).
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Errors appear in my log when I start Home Assistant

### Reproduction steps

1. Just start Home Assistant

Github: There was an error creating your issue: body is too long (maximum is 65536 characters). 

### Debug logs

```text
2024-02-09 16:02:58.011 ERROR (MainThread) [homeassistant.components.button] Error while setting up battery_notes platform for button
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/button.py", line 124, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/button.py", line 72, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '2bf7c700067b1647e28d888f191c0315'
2024-02-09 16:02:58.018 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up battery_notes platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/sensor.py", line 154, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/sensor.py", line 106, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '2bf7c700067b1647e28d888f191c0315'

2024-02-09 16:02:58.026 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up battery_notes platform for binary_sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/binary_sensor.py", line 132, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/binary_sensor.py", line 80, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '2bf7c700067b1647e28d888f191c0315'

2024-02-09 16:02:58.046 ERROR (MainThread) [homeassistant.components.button] Error while setting up battery_notes platform for button
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/button.py", line 124, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/button.py", line 72, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '483ba32b952e5060a3c6c77c93b14810'
2024-02-09 16:02:58.049 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up battery_notes platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/sensor.py", line 154, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/sensor.py", line 106, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '483ba32b952e5060a3c6c77c93b14810'

2024-02-09 16:02:58.051 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up battery_notes platform for binary_sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 344, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/battery_notes/binary_sensor.py", line 132, in async_setup_entry
    device_id = async_add_to_device(hass, config_entry)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/battery_notes/binary_sensor.py", line 80, in async_add_to_device
    device_registry.async_update_device(device_id, add_config_entry_id=entry.entry_id)
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 655, in async_update_device
    old = self.devices[device_id]
          ~~~~~~~~~~~~^^^^^^^^^^^
  File "/usr/local/lib/python3.12/collections/__init__.py", line 1134, in __getitem__
    raise KeyError(key)
KeyError: '483ba32b952e5060a3c6c77c93b14810'
```

**Comment by andrew-codechimp** [comment]
Seems to be an issue where the battery note for a device still exists, but the device itself has been deleted.
Check there aren't any lingering battery notes devices where you've deleted the original device.
I'll put a fix in to handle this.

**Comment by andrew-codechimp** [comment]
This is now fixed in the just release 2.0.12
